### PR TITLE
feat: vue-cli move in vite

### PR DESCRIPTION
**Pull request by zhangmo8**
## Why are the changes needed?

Vue-Cli is too slow.So I migrated from vue-cli to vite

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/dashboard/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        'dark-bg-color': '#202a40',
+        'dark-bg-primary-color': '#1a2232'
+      },
+      javascriptEnabled: true
+    }
+  }
+}
+
+// https://vitejs.dev/config/
+export default defineConfig({
```
Comment (minteliuwm) [inline ams/dashboard/vite.config.ts:29]: We need to build the static resources into the `/src/main/resources/static/ `directory, and we need to commit the packaged static resources, so we may need to configure the `build.outDir`

File: ams/dashboard/src/utils/request.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import useStore from '@/store'
 import router from '@/router'
 
 // http://10.196.98.23:19111/
-export const baseURL = '/'
+export const baseURL = '/mock'
```
Comment (minteliuwm) [inline ams/dashboard/src/utils/request.ts:26]: I don't think it's necessary to change the baseUrl to `/mock` by default here, maybe add a comment line, if you need to use the mock service, you can change the baseUrl to `/mock`.

File: ams/dashboard/package.json [hunk 3]
```diff
@@ -1,68 +1,47 @@
...
   "private": true,
+  "version": "0.0.0",
   "scripts": {
-    "serve": "vue-cli-service serve",
-    "build": "vue-cli-service build",
-    "build:vue-dev": "vue-cli-service build --mode dev",
-    "build:vue-test": "vue-cli-service build --mode test",
-    "build:vue-prod": "vue-cli-service build --mode prod",
-    "test:unit": "vue-cli-service test:unit",
-    "lint": "vue-cli-service lint"
```
Comment (minteliuwm) [inline ams/dashboard/package.json:12]: Maybe we need a new lint command and need to change the pre-commit at the end of the file.

File: ams/dashboard/vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  plugins: [
+    vue(),
+    viteMockServe({
+      mockPath: 'mock',
+      enable: true,
+    })
+  ],
+  server: {
+    port: 8080,
+    // proxy: {
```
Comment (minteliuwm) [inline ams/dashboard/vite.config.ts:40]: I wonder if a comment could be added here to explain how to use the proxy config?

**Review by minteliuwm — COMMENTED** [review]
Would it be possible to update the README with a description of how to install and use pnpm, as well as a description of how to use mock and how to connect to a real server, to make it a bit more friendly to other developers?

**Review by minteliuwm — APPROVED** [review]
LGTM

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by HuangFru** [comment]
After this PR, I can not log in.
image

**Comment by zhangmo8** [comment]
Hi  @HuangFru 

If you are using a local service, you need to change the configuration according to the local service. For details, you can view the [README](https://github.com/NetEase/amoro/blob/master/ams/dashboard/README.md).

 If you are not using a local service, you can wait for the mock api to be connected. 

Then now mock api is too different from the current backend real interface definition.

 So PR Welcome
